### PR TITLE
Adds toggle_universal_link

### PR DESCRIPTION
**Pull request by manny-m**
### [1 of 2 PRs to close 5177](https://openedx.atlassian.net/browse/ENT-5177)

This PR adds the ability to easily toggle the `enable_universal_link` feature in the `EnterpriseCustomer` model. 

Toggling this has side affects as outlined in the ticket.

Test are WIP

**Merge checklist:**
- [ ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ ] `make static` has been run to update webpack bundling if any static content was updated
- [ ] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the lms/edx-enterprise requirements installed or if you checked out edx-enterprise into the src directory used by lms, you can run this command through an lms shell.
        - It would be `./manage.py lms makemigrations` in the shell.
- [ ] [Version](https://github.com/edx/edx-enterprise/blob/master/enterprise/__init__.py) bumped
- [ ] [Changelog](https://github.com/edx/edx-enterprise/blob/master/CHANGELOG.rst) record added
- [ ] Translations updated (see docs/internationalization.rst but also this isn't blocking for merge atm)

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/edx-enterprise/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [GitHub Actions](https://github.com/edx/edx-enterprise/actions), verify version has been pushed to [PyPI](https://pypi.org/project/edx-enterprise/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [edx-platform](https://github.com/edx/edx-platform) to upgrade dependencies (including edx-enterprise)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-platform will look for the latest version in PyPi.
    - Note: the edx-enterprise constraint in edx-platform **must** also be bumped to the latest version in PyPi.

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -17,6 +17,10 @@ Unreleased
 ----------
 * Nothing
 
+[3.37.00]
```
Comment (iloveagent57) [inline CHANGELOG.rst:20]: Nit: we typically use single '0's.

File: enterprise/models.py [hunk 2]
```diff
@@ -713,6 +713,37 @@ def notify_enrolled_learners(self, catalog_api_user, course_id, users, admin_enr
...
+
+        # If universal link is being disabled
+        if not enable_universal_link:
+            # Deactivate all EnterpriseCustomerInviteKey
+            EnterpriseCustomerInviteKey.objects.filter(
+                enterprise_customer=self,
+                is_active=True,
+            ).update(is_active=False)
+        # If universal link is being enabled and a date is passed
+        elif bool(link_expiration_date):
```
Comment (iloveagent57) [inline enterprise/models.py:740]: I think you need to check if any `EnterpriseCustomerInviteKeys` already exist when it's being toggled to `True` - if they do, we wouldn't want to create a new one, right?

**Review by iloveagent57 — COMMENTED** [review]
This is looking good so far, minor suggestion/addition to the business logic.

File: enterprise/models.py (hunk 2 above)
Comment (long74100) [inline enterprise/models.py:740]: I think we can do ```elif link_expiration_date``` here.

File: enterprise/models.py [hunk 3]
```diff
@@ -713,6 +713,37 @@ def notify_enrolled_learners(self, catalog_api_user, course_id, users, admin_enr
...
         send_enterprise_email_notification.delay(self.uuid, admin_enrollment, email_items)
 
+    def toggle_universal_link(self, enable_universal_link, link_expiration_date):
+        """
+        Sets enable_universal_link
+
+        If there is no change to be made, return
+
+        When enable_universal_link changes to;
+            True: a new link is created
```
Comment (long74100) [inline enterprise/models.py:723]: should we reference customer invite key instead of links here?

File: enterprise/models.py (hunk 2 above)
Comment (iloveagent57) [reply enterprise/models.py:740]: Ah, that's right - the links active state is immutable!  So yeah, leave what you have. 👍

**Pull request merged**